go-controller/pkg/ovn/controller/admin_network_policy/admin_network_policy_controller.go

```diff
@@ -0,0 +1,469 @@
+package adminnetworkpolicy
+
+import (
+	"fmt"
+	"reflect"
+	"sync"
+	"time"
+
+	libovsdbclient "github.com/ovn-org/libovsdb/client"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/factory"
+	addressset "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/address_set"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+	anpapi "sigs.k8s.io/network-policy-api/apis/v1alpha1"
+	anpclientset "sigs.k8s.io/network-policy-api/pkg/client/clientset/versioned"
+	anplister "sigs.k8s.io/network-policy-api/pkg/client/listers/apis/v1alpha1"
+)
+
+const (
+	// maxRetries is the number of times a object will be retried before it is dropped out of the queue.
+	// With the current rate-limiter in use (5ms*2^(maxRetries-1)) the following numbers represent the
+	// sequence of delays between successive queuings of an object.
+	//
+	// 5ms, 10ms, 20ms, 40ms, 80ms, 160ms, 320ms, 640ms, 1.3s, 2.6s, 5.1s, 10.2s, 20.4s, 41s, 82s
+	maxRetries = 15
+
+	controllerName = "admin-network-policy-controller"
```

Comment:
oh yes I am using this controller name for dbIDs I am using this as my `objectIDs.ownerControllerName` for all the objects that are owned by this controller. 

not sure if I follow your suggestion here, might need some clarification:

```
I think it may be better to pass default/secondary controller name to the Controller, since that is the initial semantics for dbIDs
```
Do you mean something like: https://github.com/ovn-org/ovn-kubernetes/pull/3308/commits/1e0c4cc8cceb03961470b9d38ea86465b8401c49#diff-214cf3919602fd060047b8d15fd6c0ca9d3ed3d42c47fff4b181a072c182b673R70 ? pass the name into the `Controller` struct so that in future its more expandable?

OR do you mean all objects created here need to be owned by default/secondary network controllers?

Reply:
ack will do I can call this `default-admin-network-policy-controller`

Reply:
wait a sec, I guess I didn't explain that well enough, so dbIndex controller name should be the name of the default or secondary controller, so you don't need a new constant, but just pass `oc.controllerName` as an argument here https://github.com/ovn-org/ovn-kubernetes/pull/3659/commits/eb67781f6132c25a647b66d07dedafb94924024a#diff-3f6ed978237c56cacb91eb947304ed9e429edf24d8d358be806093b0f061e154R440

does it make sense?

Reply:
OH dangit now I get it.
SO: `libovsdbops.OwnerControllerKey` is either default-network-controller OR secondary-network-controller or tertiary-network-controller or 100th-network-controller and `libovsdbops.OwnerTypeKey` is what will signal its being owned by ANP for me. gotcha. ack that makes sense, seems like egressSVC is doing this  as well..
(we should add docs for what owner-controller and owner-type etc mean)

Reply:
So since 2ndary networks don't support this feature yet, I will for now do same as this: https://github.com/ovn-org/ovn-kubernetes/blob/feea09c6d0c2dd47e6d13136fe6e3d7b015dfa66/go-controller/pkg/ovn/ovn.go#L430 and when we do support 2ndary/3rdary or whatever in future we can just remove that constant and expand that.

Reply:
leaving the comment open till @npinaeva : you confirm if that is ok ? :)

Reply:
yeah looks good!